x-pack/plugins/cases/common/api/user.ts

```diff
@@ -7,19 +7,22 @@
 
 import * as rt from 'io-ts';
 
+const UserWithoutProfileUidRt = rt.type({
+  email: rt.union([rt.undefined, rt.null, rt.string]),
+  full_name: rt.union([rt.undefined, rt.null, rt.string]),
+  username: rt.union([rt.undefined, rt.null, rt.string]),
+});
+
 export const UserRt = rt.intersection([
-  rt.type({
-    email: rt.union([rt.undefined, rt.null, rt.string]),
-    full_name: rt.union([rt.undefined, rt.null, rt.string]),
-    username: rt.union([rt.undefined, rt.null, rt.string]),
-  }),
+  UserWithoutProfileUidRt,
   rt.partial({ profile_uid: rt.string }),
 ]);
 
 export const UserWithProfileInfoRt = rt.intersection([
   rt.type({
-    user: UserRt,
+    user: UserWithoutProfileUidRt,
   }),
+  rt.partial({ uid: rt.string }),
```

Comment:
I wonder if there's a way we can tie this implementation with the security plugin 🤔 My concern is that if they make a breaking change to the response structure and make the changes throughout the code base, this won't show up as needing to be changed and it will break the UI.

Here's an idea, how about we add an integration test that does a bulk get using the security API and then does a decode using this schema? Or something like that, that way if there's a breaking change at least we'll get a failing test.

Reply:
Amazing idea! I was thinking about the same but I could not find a good way to do it.